### PR TITLE
cWindow: Convert XYZ to Vector3

### LINKED ISSUE #4675: cWindow: Replace XYZ coords with Vector3
Replace all 3-coord parameters with `Vector3`-based parameters in the `cWindow` and descendants.

This needs to wait until #4673 and #4674 are finished.

### PR DESCRIPTION
**Pull request by MaxwellScroggs**
Converted the depreciated X,Y,Z coordinate format into the newer Vector3 format for cWindow and all of its decendants.
Style issues resolved.
Fixes #4675

### CODE REVIEW

**Comment by madmaxoft** [comment]
Welcome to the project.
You don't need to close a PR and open a new one when you change things - just add more commits to the branch in your repo and this PR will be automatically updated.
(This also works in the other, bad way - now you cannot add other *unrelated* commits to the branch without them showing in this PR)

File: .gitignore [hunk 1]
```diff
@@ -121,3 +121,4 @@ build-cuberite
 # clang-tidy
 tidy-build
 run-clang-tidy.py
+/VS2019-x64
```
Comment (madmaxoft) [inline .gitignore:124]: You should put this into your local (per-machine) gitignore file - this is a folder name chosen by you, not by anything in the repo, and so is unrelated to the code here.

File: src/UI/AnvilWindow.cpp [hunk 2]
```diff
@@ -49,11 +47,9 @@ void cAnvilWindow::SetRepairedItemName(const AString & a_Name, cPlayer * a_Playe
 
 
 
-void cAnvilWindow::GetBlockPos(int & a_PosX, int & a_PosY, int & a_PosZ)
+void cAnvilWindow::GetBlockPos(Vector3i & a_Pos)
```
Comment (madmaxoft) [inline src/UI/AnvilWindow.cpp:50]: Ideally, this function should now return the `Vector3i`
```suggestion
const Vector3i & cAnvilWindow::GetBlockPos()
```

File: src/UI/SlotArea.cpp [hunk 3]
```diff
@@ -996,14 +996,14 @@ void cSlotAreaAnvil::OnTakeResult(cPlayer & a_Player)
...
+	Vector3i Pos;
+	static_cast<cAnvilWindow &>(m_ParentWindow).GetBlockPos(Pos);
 
 	BLOCKTYPE Block;
 	NIBBLETYPE BlockMeta;
-	a_Player.GetWorld()->GetBlockTypeMeta(PosX, PosY, PosZ, Block, BlockMeta);
+	a_Player.GetWorld()->GetBlockTypeMeta(Pos, Block, BlockMeta);
 
-	const Vector3i BlockPos{PosX, PosY, PosZ};
+	const Vector3i BlockPos{Pos};
```
Comment (madmaxoft) [inline src/UI/SlotArea.cpp:1006]: This copy is not needed now, just rename `Pos` to `BlockPos` in line 999, then use it everywhere.

File: src/UI/CraftingWindow.cpp [hunk 4]
```diff
@@ -10,7 +10,7 @@
 
 
 
-cCraftingWindow::cCraftingWindow(int a_BlockX, int a_BlockY, int a_BlockZ) :
+cCraftingWindow::cCraftingWindow(Vector3i a_Block) :  // This Vector3i is never saved or used? Consider removing
```
Comment (madmaxoft) [inline src/UI/CraftingWindow.cpp:13]: Then do remove it now - this refactoring is the ideal time for that.

File: src/UI/SlotArea.cpp [hunk 5]
```diff
@@ -1407,11 +1407,9 @@ void cSlotAreaBeacon::OnSlotChanged(cItemGrid * a_ItemGrid, int a_SlotNum)
 ////////////////////////////////////////////////////////////////////////////////
 // cSlotAreaEnchanting:
 
-cSlotAreaEnchanting::cSlotAreaEnchanting(cWindow & a_ParentWindow, int a_BlockX, int a_BlockY, int a_BlockZ) :
+cSlotAreaEnchanting::cSlotAreaEnchanting(cWindow & a_ParentWindow, Vector3i a_Block) :
```
Comment (madmaxoft) [inline src/UI/SlotArea.cpp:1410]: Use `a_BlockPos` (and `m_BlockPos`) names for increased code clarity.

File: src/UI/SlotArea.cpp [hunk 6]
```diff
@@ -1644,7 +1642,7 @@ int cSlotAreaEnchanting::GetBookshelvesCount(cWorld & a_World)
 {
 	int Bookshelves = 0;
 	cBlockArea Area;
-	Area.Read(a_World, m_BlockX - 2, m_BlockX + 2, m_BlockY, m_BlockY + 1, m_BlockZ - 2, m_BlockZ + 2);
+	Area.Read(a_World, m_Block.x - 2, m_Block.x + 2, m_Block.y, m_Block.y + 1, m_Block.z - 2, m_Block.z + 2);
```
Comment (madmaxoft) [inline src/UI/SlotArea.cpp:1645]: This can now use the `Vector3i`-based `Read` function, which is better (safer).

**Comment by madmaxoft** [comment]
Overall well done, just minor issues.

File: src/UI/AnvilWindow.cpp (hunk 2 above)
Comment (madmaxoft) [reply src/UI/AnvilWindow.cpp:50]: Could be even moved into the header file now.

File: CONTRIBUTORS [hunk 7]
```diff
@@ -69,3 +69,4 @@ worktycho
 Xenoxis
 xoft (Mattes Dolak/madmaxoft on GH)
 Yeeeeezus (Donated AlchemistVillage prefabs)
+MaxwellScroggs
```
Comment (madmaxoft) [inline CONTRIBUTORS:72]: Please keep the list sorted

**Comment by madmaxoft** [comment]
Sorry for being a nit-picker :)

**Comment by Electric26** [comment]
> Sorry for being a nit-picker :)

Personally, I really like it. It results in quality code. I sure wouldn't want to contribute to a project with (unusably) bad code.

**Pull request merged**